pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -255,15 +255,15 @@ func (r *shootReconciler) runDeleteShootFlow(ctx context.Context, o *operation.O
 			Fn:           flow.TaskFn(botanist.DeployEtcd).RetryUntilTimeout(defaultInterval, defaultTimeout).DoIf(cleanupShootResources),
 			Dependencies: flow.NewTaskIDs(deploySecrets, deployCloudProviderSecret, deployOwnerDomainDNSRecord),
 		})
-		_ = g.Add(flow.Task{
+		scaleETCD = g.Add(flow.Task{
```

Comment:
I'm wondering, why we need a separate task to scale up etcd at all. Can't we simply make the deploy etcd task scale it up in any case during the shoot deletion flow (always set `spec.replicas=1`)?

I.e. we would need to adapt https://github.com/gardener/gardener/blob/1e09e7d09543d0d4c96207b18703187c400b0e1d/pkg/operation/botanist/etcd.go#L58 to be `b.Shoot.HibernationEnabled && b.Shoot.GetInfo().DeletionTimestamp == nil`.
This will set `retainReplicas` to `false` for shoots in deletion, which will deploy etcd with `spec.replicas=1`, IISIC: https://github.com/gardener/gardener/blob/1e09e7d09543d0d4c96207b18703187c400b0e1d/pkg/operation/botanist/component/etcd/etcd.go#L698-L700

Reply:
I wondered about this as well but decided to focus on the fix right now, i.e. to not change the deletion logic. Can we tackle this separately?

Reply:
Yes, fair enough. Let's tackle it separately.